core/include/thread_flags.h

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (C) 2015 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     core_thread
+ * @brief       Thread flags
+ * @{
+ *
+ * @file
+ * @brief       Thread flags API
+ *
+ * This API can be used to notify threads of conditions in a race-free
+ * and allocation-less way.
+ *
+ * Each thread has a number boolean flags (usually 16) in its tcb, stored in
```

Comment:
Well, I think it is a little bit confusing, because right now it is kind of fixed to 16 flags unless you change the typedef what I would not count as "configurable". See what I mean?